test/e2e/node_os_test.go

```diff
@@ -22,11 +31,16 @@ const (
 // Verifies that the OS on each node is identifiable using the mechanism the
 // MCD does. This is done to ensure that we get an early warning if the
 // contents of /etc/os-release or /usr/lib/os-release unexpectedly change.
+//
+// We do the same for the labels attached to the OS image associated with the
+// release. This is done for the same early warning.
 func TestOSDetection(t *testing.T) {
 	cs := framework.NewClientSet("")
 	nodes, err := cs.CoreV1Interface.Nodes().List(context.TODO(), metav1.ListOptions{})
 	require.NoError(t, err)
 
+	osImageRelease, osImageReleaseSrc := getOSReleaseInfoForOSImage(t, cs, nodes.Items[0])
```

Comment:
I added an additional check to this e2e test that gets the OS image labels given the current release on the test cluster. This way, we can get the same early warning if something changes unexpectedly.